debugger/src/main/scala/org.apache.daffodil.debugger.dap/DAPodil.scala

```diff
@@ -748,11 +748,24 @@ object DAPodil extends IOApp {
       copy(value = value + (uri.normalize -> lines))
 
     def contains(location: Location): Boolean =
-      value.exists {
-        // format: off
-        case (uri, lines) =>
-          uri == location.uri && lines.exists(_ == location.line)
-        // format: on
+      value.exists { case (uri, lines) =>
```

Comment:
@mbeckerle Okay so would something like `debugger/src/main/scala/org.apache.daffodil.debugger.dap/Portability.scala` work for that file? I believe this will be permanent work around because the `location.uri` is what produces the URI like `file:/Users/user/path.file`, which I believe should be created/initialized in the Scala somewhere as `java.net.URI`. But the URI passed in from the extension is coming in like `file:///Users/user/path.file`.

However, I could also use `getPath` or `getRawPath` to use for comparisons instead. This would eliminate the need for caring about if the URI starts with `file://` or `file:`, also eliminating the `replace("file://", "file:")`. But in the case of windows it would still need the fix of making the drive letter uppercase.

This fix may be needed else where but I would need to do more thorough check to know for sure.

Reply:
So the line that may cause the issue is this:

```scala
URI.create(loc.uriString).normalize
```

Whenever I get `loc.uriString` value it looks like `file:/Users/user/path.file`. So it is a problem between the Extension and Debugger starting the URI differently. Then for windows there is another is issue where one URI is using uppercase and the other one is using lowercase for the drive letter.

Reply:
@arosien Yes the main issue that one URI used an uppercase drive letter and the other used lowercase. So if we just compare the paths we avoid any issues with the different way that the URI's start with `file:`, which may not be an issue any way I am not sure. But we want to use the paths so we know the index of where the drive letter will be in a split array. Basically, to fix the drive letter issue we need to split the path and either uppercase or lowercase the drive letter. I think normally uppercase is Windows default so that is what I planned to use.

Reply:
Here is some extra context:
- Daffodil provides schema locations [as a `String` that is a URL](https://github.com/apache/daffodil/blob/d5e084c34abbd337d0951a6d33460465e38692e8/daffodil-lib/src/main/scala/org/apache/daffodil/lib/exceptions/SchemaFileLocatable.scala#L131-L148), although other code assumes this URL string is URL-encoded. It would be nice if the field was actually an `URL`.
- Because Daffodil provides a URL-string, we convert the paths sent via DAP to URIs, and normalize both DAP- and Daffodil-provided URIs. (We could instead convert the Daffodil URL-string to a `Path` and not convert the DAP paths to URIs, but that seemed more error prone, or at least less "universal". Except URI normalization doesn't seem to cover Windows drive letters. But FYI `Path` does handle this, or at least purports to via `getAbsolutePath`.)

Reply:
@mbeckerle Would the suggested changes from https://github.com/apache/daffodil-vscode/pull/990#discussion_r1521780315 be good to to fix the issue and get this PR merged?

Reply:
> @mbeckerle Would the suggested changes from [#990 (comment)](https://github.com/apache/daffodil-vscode/pull/990#discussion_r1521780315) be good to to fix the issue and get this PR merged?

Hey @mbeckerle just checking back in on this PR as the changes you mentioned will be the main thing holding it from getting merged. So will the fix I mentioned work?